compiler/src/dotty/tools/dotc/config/ScalaSettings.scala

```diff
@@ -23,7 +23,20 @@ object ScalaSettingCategories:
   val AdvancedSetting = "X"
   val VerboseSetting = "V"
 
-object ScalaSettings:
+type ScalaSettings = ScalaSettings.type
```

Comment:
Why do you need this?

Reply:
I didn't want to have `.type` everywhere ScalaSettings are stored, I'd rather have it as a type. But it kind of gives a false impression of being instantiable and is not a widely used pattern here. And is not really necessary in the end, we don't need an instance of it as value, as it's now immutable.